app/commands/thredded/mark_all_read.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+module Thredded
+  class MarkAllRead
+    def initialize(topic_type, user)
+      @topic_type = topic_type
+      @user = user
+    end
+
+    def run
+      @topic_type.unread(@user).each do |topic|
```

Comment:
What do you think about something like ... 

```ruby
unread_topics = @topic_type.unread(@user)
return if unread_topics.empty?

unread_topics.each do |topic|
  # ... 
```

Using a guard clause could bail out from the rest of everything nice and early. It's a negligible benefit since the enumerator will for all intents and purposes never loop through an empty collection but it still reads pretty well.